dms/models/directory.py

```diff
@@ -76,10 +70,25 @@ class DmsDirectory(models.Model):
         index=True,
         store=True,
         readonly=False,
-        compute="_compute_directory_type",
         copy=True,
+        default=lambda self: self._default_parent_id(),
     )
 
+    def _default_parent_id(self):
+        if self._context.get("active_model") == self._name and self._context.get(
```

Comment:
`self._context` is an internal variable you shouldn't use. Cache the value:
```python
context = self.env.context
```
and then use it across the code.